servicebus/data-plane/azure-servicebus/src/main/java/com/microsoft/azure/servicebus/primitives/AsyncUtil.java

```diff
@@ -6,53 +6,51 @@
 import java.util.concurrent.Callable;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ExecutionException;
+import java.util.concurrent.RejectedExecutionException;
 
 // To complete futures using a different thread. Otherwise every future is completed on the single reactor thread
 // which badly affects perf and a client can potentially kill the thread or lock the thread.
 class AsyncUtil {
 
-    public static <T> boolean completeFutureAndGetStatus(CompletableFuture<T> future, T result)
-    {
+    public static <T> boolean completeFutureAndGetStatus(CompletableFuture<T> future, T result) {
         try {
-            return MessagingFactory.INTERNAL_THREAD_POOL.submit(new CompleteCallable<T>(future, result)).get();
+            return MessagingFactory.INTERNAL_THREAD_POOL.submit(new CompleteCallable<>(future, result)).get();
         } catch (InterruptedException | ExecutionException e) {
             e.printStackTrace();
             return false;
         }
     }
 
-    public static <T> void completeFuture(CompletableFuture<T> future, T result)
-    {
-        MessagingFactory.INTERNAL_THREAD_POOL.submit(new CompleteCallable<T>(future, result));
+    public static <T> void completeFuture(CompletableFuture<T> future, T result) {
+        MessagingFactory.INTERNAL_THREAD_POOL.submit(new CompleteCallable<>(future, result));
     }
 
-    public static <T> boolean completeFutureExceptionallyAndGetStatus(CompletableFuture<T> future, Throwable exception)
-    {
+    public static <T> boolean completeFutureExceptionallyAndGetStatus(CompletableFuture<T> future, Throwable exception) {
         try {
-            return MessagingFactory.INTERNAL_THREAD_POOL.submit(new CompleteExceptionallyCallable<T>(future, exception)).get();
+            return MessagingFactory.INTERNAL_THREAD_POOL.submit(new CompleteExceptionallyCallable<>(future, exception)).get();
         } catch (InterruptedException | ExecutionException e) {
             e.printStackTrace();
             return false;
         }
     }
 
-    public static <T> void completeFutureExceptionally(CompletableFuture<T> future, Throwable exception)
-    {
-        MessagingFactory.INTERNAL_THREAD_POOL.submit(new CompleteExceptionallyCallable<T>(future, exception));
+    public static <T> void completeFutureExceptionally(CompletableFuture<T> future, Throwable exception) {
+        MessagingFactory.INTERNAL_THREAD_POOL.submit(new CompleteExceptionallyCallable<>(future, exception));
     }
 
-    public static void run(Runnable runnable)
-    {
-        MessagingFactory.INTERNAL_THREAD_POOL.submit(runnable);
+    public static void run(Runnable runnable) {
+        try {
```

Comment:
[SpotBugs-P3]
Exceptional return value of java.util.concurrent.ExecutorService.submit(Runnable) ignored in com.microsoft.azure.servicebus.primitives.AsyncUtil.run(Runnable)

Reply:
Is there any logger? Should check with the service team to ask if this behaviour change is OK with them.

Reply:
No longer an issue. Will suppress it.